apps/golden-sample-app/src/app/app.component.spec.ts

```diff
@@ -1,11 +1,19 @@
+import { LayoutService } from '@backbase/ui-ang';
+import { OAuthService } from 'angular-oauth2-oidc';
 import { AppComponent } from './app.component';
 describe('AppComponent', () => {
   let component: AppComponent;
-  const mockOAuthService: any = {};
-  const mockLayoutService: any = {};
+  const mockOAuthService: Pick<OAuthService, 'logOut' | 'hasValidAccessToken'> =
```

Comment:
I think this typing is a step in the right direction. However if I (for instance) want to write this test, it leads to an error

```  it('should do foo for invalid access token', () => {
    mockOAuthService.hasValidAccessToken.mockReturnValueOnce(false);
    
    // expect...
  });
```
solution could be to use `const mockOAuthService: jest.Mocked<Pick<OAuthService, 'logOut' | 'hasValidAccessToken'>>`

but again, curious about other opinions on this

Reply:
In this case you would either have to do something like this
```
component = new AppComponent(
      mockOAuthService as unknown as OAuthService,
      mockLayoutService
    );
```

or use Testbed of course

Reply:
It's testing code, I'm not too concerned in using any for mocks and spies. What I do normally is using actual implementations and stub only the methods I need. 

For what concerns TestBed, removing it forces you to use full-on mocks for all dependencies, so sometimes it might be a bit easier to just rely on that and actual stubbed/spied real classes.

Reply:
Okay so while it is not necessarily my opinion :) the conclusion seems to be:

- using `any` for mocks is fine
- we use Testbed

correct @santam85 ?

Reply:
It depends on the constructor of the dependency being used to determine whether mocks or a new instance should be made. If playwright will be handling the e2e, focusing purely on the logic in the components is more effective. 

@jordydejong If you check locale-selector.component.spec.ts you can see how it's possible to test an invalid token in your example. This approach works works and is needed in cases anyways for any values set on initialization. 
That being said, I see the value of jest.Mocked now and will add it if we stick with this approach.

@santam85 
If the final decision is to use TestBed,  I will revert all the changes to the previous unit tests and setup Jasmine/Karma in Nx. If we use TestBed having createSpyObj is an important piece.